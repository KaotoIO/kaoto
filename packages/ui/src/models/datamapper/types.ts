/**
 * The data types mapped from XML Schema +.
 */
export enum Types {
  AnyAtomicType = 'anyAtomicType',
  AnyType = 'anyType',
  AnyURI = 'anyURI',
  String = 'string',
  Float = 'float',
  Double = 'double',
  Integer = 'integer',
  Decimal = 'decimal',
  Boolean = 'boolean',
  Date = 'date',
  DateTime = 'dateTime',
  Time = 'time',
  QName = 'QName',
  Duration = 'duration',
  DayTimeDuration = 'dayTimeDuration',
  NCName = 'NCName',
  // not XSD predefined but appear in XPath function signature
  Numeric = 'numeric',
  Item = 'item()',
  Element = 'element()',
  Node = 'node()',
  DocumentNode = 'document-node()',
  // custom types
  Container = 'Container',
  Array = 'Array',
}
