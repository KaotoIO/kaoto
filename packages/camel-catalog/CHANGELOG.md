# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.3.0 (2023-11-22)

### Bug Fixes

* **deps:** update dependency io.fabric8:kubernetes-model to v6.9.0 ([1923d4d](https://github.com/KaotoIO/kaoto-next/commit/1923d4d09a1d4122eb0455f937221205a27b4cd1))
* **deps:** update dependency io.fabric8:kubernetes-model to v6.9.1 ([f812b71](https://github.com/KaotoIO/kaoto-next/commit/f812b7180688769308c7551085fccc714f1034ac))
* **deps:** update dependency io.fabric8:kubernetes-model to v6.9.2 ([419c60f](https://github.com/KaotoIO/kaoto-next/commit/419c60fffa034a0eae01624d1e891ea525224407))
* **deps:** update dependency org.apache.maven.plugin-tools:maven-plugin-annotations to v3.10.1 ([4157faa](https://github.com/KaotoIO/kaoto-next/commit/4157faa8426cdf7a6346cc9558f3674a73b0836e))
* **deps:** update dependency org.apache.maven.plugin-tools:maven-plugin-annotations to v3.10.2 ([df62ce2](https://github.com/KaotoIO/kaoto-next/commit/df62ce2848e6beeafc5254f4e2acfc89ae65bf8c))
* **deps:** update version.jackson to v2.15.3 ([c56e2a6](https://github.com/KaotoIO/kaoto-next/commit/c56e2a66db78c026d7707eded2e0c3d4439837a3))
* **deps:** update version.jackson to v2.16.0 ([6f1910b](https://github.com/KaotoIO/kaoto-next/commit/6f1910ba6cc241d0fa932c1a3b1119ec80cb2b49))
* Refactor @kaoto-next/camel-catalog ([e4796d2](https://github.com/KaotoIO/kaoto-next/commit/e4796d270a61d41615d7cdf5ea7dc2f396f6640a)), closes [#311](https://github.com/KaotoIO/kaoto-next/issues/311) [#316](https://github.com/KaotoIO/kaoto-next/issues/316) [#335](https://github.com/KaotoIO/kaoto-next/issues/335) [#311](https://github.com/KaotoIO/kaoto-next/issues/311)
* Support using camel catalog and schema in Jest tests ([365f7a0](https://github.com/KaotoIO/kaoto-next/commit/365f7a0172f3cbfeb9ea2f5c6b7c4074003bf75c)), closes [#310](https://github.com/KaotoIO/kaoto-next/issues/310)

### Features

* **catalog:** Minify Camel Catalog ([5c17105](https://github.com/KaotoIO/kaoto-next/commit/5c1710544af50f4f040ef7dcf94d69f801af2170))
* Support configuring /PipeSpec/ErrorHandler ([1cd996f](https://github.com/KaotoIO/kaoto-next/commit/1cd996fb53a89ef41b23ba4b84c7b2e27095516f)), closes [#188](https://github.com/KaotoIO/kaoto-next/issues/188)
* Support configuring Camel K CR metadata ([#181](https://github.com/KaotoIO/kaoto-next/issues/181)) ([#226](https://github.com/KaotoIO/kaoto-next/issues/226)) ([31aba1a](https://github.com/KaotoIO/kaoto-next/commit/31aba1a9d2395665d2735f2f3740d300f638e98f))
* Support configuring dataformat in Canvas form ([2a62d04](https://github.com/KaotoIO/kaoto-next/commit/2a62d04d070826f18bb7c5a01a3fd9ab2dd83534)), closes [#273](https://github.com/KaotoIO/kaoto-next/issues/273)

# 0.2.0 (2023-10-05)

### Bug Fixes

* **build:** Build errors with node v20 ([1cdc56f](https://github.com/KaotoIO/kaoto-next/commit/1cdc56fe96e3e46a1ac84e3423193c6d5c6406dd))
* **build:** Windows build ([b8d962e](https://github.com/KaotoIO/kaoto-next/commit/b8d962e0b86e4854277d354c53185a23e2ff9ed9))
* Camel K CRD file extraction fails ([c633b07](https://github.com/KaotoIO/kaoto-next/commit/c633b07c419fad48f0e4b65b3f4a0687819aa4ba)), closes [#106](https://github.com/KaotoIO/kaoto-next/issues/106)
* Create sub schemas for each top properties ([#144](https://github.com/KaotoIO/kaoto-next/issues/144)) ([40efd4a](https://github.com/KaotoIO/kaoto-next/commit/40efd4a18af9e4da20394943979f5c2a58fe8e8f)), closes [#138](https://github.com/KaotoIO/kaoto-next/issues/138)
* **deps:** update dependency io.fabric8:kubernetes-model to v6.8.1 ([#32](https://github.com/KaotoIO/kaoto-next/issues/32)) ([e38f7ab](https://github.com/KaotoIO/kaoto-next/commit/e38f7abaca83504927ac5e5bee16f9a80f5a27f4))
* **deps:** update dependency org.apache.maven:maven-plugin-api to v3.9.5 ([#191](https://github.com/KaotoIO/kaoto-next/issues/191)) ([598fd26](https://github.com/KaotoIO/kaoto-next/commit/598fd26665b08cba3692685dd5ac158245fc2fde))
* Let UI use maven generated YAML DSL sub schema ([66bdd77](https://github.com/KaotoIO/kaoto-next/commit/66bdd7704c87d2d29c602d36557f10d5ddba5ba5)), closes [#147](https://github.com/KaotoIO/kaoto-next/issues/147) [#8](https://github.com/KaotoIO/kaoto-next/issues/8)
* Put schema files into a Map to allow direct access with a schema name ([9cf8c4d](https://github.com/KaotoIO/kaoto-next/commit/9cf8c4d538908732b23198d394e0dcfa95f283ff)), closes [#148](https://github.com/KaotoIO/kaoto-next/issues/148)
* Remove kebab-case YAML schema in favor of camelCase one ([96b21e4](https://github.com/KaotoIO/kaoto-next/commit/96b21e48d1b483ce0dae46d6b28a53c02d1ebdad)), closes [#70](https://github.com/KaotoIO/kaoto-next/issues/70)
* Replace rm with rimraf to clean dist folder ([8133563](https://github.com/KaotoIO/kaoto-next/commit/81335630d329ea6b0e7c21a1d3c045ed89ff4e00))

### Features

* Add camel catalog and schema ([bb8f3ad](https://github.com/KaotoIO/kaoto-next/commit/bb8f3ad9dadfe6fe71f22fe8dbb1986d23830986))
* **camel-catalog:** export Typescript definitions from Camel Schemas ([a76985c](https://github.com/KaotoIO/kaoto-next/commit/a76985c78871f70ff8a15af97afe53970c26fc78))
* **versioning:** Version and publish workflow ([98f9587](https://github.com/KaotoIO/kaoto-next/commit/98f95879076cd066d1e0cb83d9556468c1277e7a))

# 0.1.0 (2023-10-05)

### Bug Fixes

* **build:** Build errors with node v20 ([1cdc56f](https://github.com/KaotoIO/kaoto-next/commit/1cdc56fe96e3e46a1ac84e3423193c6d5c6406dd))
* **build:** Windows build ([b8d962e](https://github.com/KaotoIO/kaoto-next/commit/b8d962e0b86e4854277d354c53185a23e2ff9ed9))
* Camel K CRD file extraction fails ([c633b07](https://github.com/KaotoIO/kaoto-next/commit/c633b07c419fad48f0e4b65b3f4a0687819aa4ba)), closes [#106](https://github.com/KaotoIO/kaoto-next/issues/106)
* Create sub schemas for each top properties ([#144](https://github.com/KaotoIO/kaoto-next/issues/144)) ([40efd4a](https://github.com/KaotoIO/kaoto-next/commit/40efd4a18af9e4da20394943979f5c2a58fe8e8f)), closes [#138](https://github.com/KaotoIO/kaoto-next/issues/138)
* **deps:** update dependency io.fabric8:kubernetes-model to v6.8.1 ([#32](https://github.com/KaotoIO/kaoto-next/issues/32)) ([e38f7ab](https://github.com/KaotoIO/kaoto-next/commit/e38f7abaca83504927ac5e5bee16f9a80f5a27f4))
* **deps:** update dependency org.apache.maven:maven-plugin-api to v3.9.5 ([#191](https://github.com/KaotoIO/kaoto-next/issues/191)) ([598fd26](https://github.com/KaotoIO/kaoto-next/commit/598fd26665b08cba3692685dd5ac158245fc2fde))
* Let UI use maven generated YAML DSL sub schema ([66bdd77](https://github.com/KaotoIO/kaoto-next/commit/66bdd7704c87d2d29c602d36557f10d5ddba5ba5)), closes [#147](https://github.com/KaotoIO/kaoto-next/issues/147) [#8](https://github.com/KaotoIO/kaoto-next/issues/8)
* Put schema files into a Map to allow direct access with a schema name ([9cf8c4d](https://github.com/KaotoIO/kaoto-next/commit/9cf8c4d538908732b23198d394e0dcfa95f283ff)), closes [#148](https://github.com/KaotoIO/kaoto-next/issues/148)
* Remove kebab-case YAML schema in favor of camelCase one ([96b21e4](https://github.com/KaotoIO/kaoto-next/commit/96b21e48d1b483ce0dae46d6b28a53c02d1ebdad)), closes [#70](https://github.com/KaotoIO/kaoto-next/issues/70)
* Replace rm with rimraf to clean dist folder ([8133563](https://github.com/KaotoIO/kaoto-next/commit/81335630d329ea6b0e7c21a1d3c045ed89ff4e00))

### Features

* Add camel catalog and schema ([bb8f3ad](https://github.com/KaotoIO/kaoto-next/commit/bb8f3ad9dadfe6fe71f22fe8dbb1986d23830986))
* **camel-catalog:** export Typescript definitions from Camel Schemas ([a76985c](https://github.com/KaotoIO/kaoto-next/commit/a76985c78871f70ff8a15af97afe53970c26fc78))
* **versioning:** Version and publish workflow ([98f9587](https://github.com/KaotoIO/kaoto-next/commit/98f95879076cd066d1e0cb83d9556468c1277e7a))
