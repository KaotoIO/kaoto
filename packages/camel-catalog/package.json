{
  "name": "@kaoto-next/camel-catalog",
  "version": "0.2.0",
  "type": "commonjs",
  "description": "Camel Catalog and schemas for Kaoto",
  "repository": "https://github.com/KaotoIO/kaoto-next",
  "repositoryDirectory": "packages/camel-catalog",
  "author": "The Kaoto Team",
  "license": "Apache License v2.0",
  "private": true,
  "main": "./package.json",
  "exports": {
    ".": "./dist",
    "./index.json": "./dist/index.json",
    "./types": "./dist/types/index.ts",
    "./package.json": "./package.json",
    "./*.json": "./dist/*.json"
  },
  "scripts": {
    "build": "yarn clean && yarn build:mvn && yarn build:ts",
    "build:mvn": "./mvnw clean install",
    "build:ts": "node --loader ts-node/esm ./src/json-schema-to-typescript.mts",
    "lint": "yarn eslint \"src/**/*.{mts,ts}\"",
    "lint:fix": "yarn lint --fix",
    "clean": "yarn rimraf ./dist"
  },
  "devDependencies": {
    "@types/node": "^18.0.0",
    "eslint": "^8.45.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-prettier": "^5.0.0",
    "json-schema-to-typescript": "^13.0.2",
    "prettier": "^3.0.0",
    "rimraf": "^5.0.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.0.2"
  }
}
