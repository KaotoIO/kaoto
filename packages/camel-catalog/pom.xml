<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>io.kaoto</groupId>
  <artifactId>kaoto-camel-catalog-parent</artifactId>
  <packaging>pom</packaging>
  <version>0.0.1-SNAPSHOT</version>
  <name>kaoto-camel-catalog-parent</name>
  <description>
    The Apache Camel catalog and YAML DSL schema collection to be used in Kaoto.
  </description>
  <url>https://kaoto.io</url>

  <repositories>
    <repository>
      <id>apache.snapshots</id>
      <name>Apache Snapshots</name>
      <url>https://repository.apache.org/content/repositories/snapshots/</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
  </repositories>
  <pluginRepositories>
    <pluginRepository>
      <id>apache.snapshots</id>
      <name>Apache Snapshots</name>
      <url>https://repository.apache.org/content/repositories/snapshots/</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </pluginRepository>
  </pluginRepositories>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <version.camel>4.6.0</version.camel>
    <version.camel-k-crds>2.3.2</version.camel-k-crds>
    <version.camel-kamelets>4.4.1</version.camel-kamelets>
    <version.jackson>2.17.1</version.jackson>
    <version.java>21</version.java>
    <version.junit>5.10.2</version.junit>
    <version.kubernetes-model>6.10.0</version.kubernetes-model>
    <version.maven-antrun-plugin>3.1.0</version.maven-antrun-plugin>
    <version.maven-compiler-plugin>3.13.0</version.maven-compiler-plugin>
    <version.maven-dependency-plugin>3.6.1</version.maven-dependency-plugin>
    <version.maven-surefire-plugin>3.2.5</version.maven-surefire-plugin>
    <version.maven-resources-plugin>3.3.1</version.maven-resources-plugin>
    <version.maven-plugin-annotations>3.11.0</version.maven-plugin-annotations>
    <version.maven-plugin-api>3.9.7</version.maven-plugin-api>
    <version.maven-plugin-plugin>3.13.1</version.maven-plugin-plugin>
  </properties>

  <modules>
    <module>kaoto-camel-catalog-maven-plugin</module>
    <module>assembly</module>
  </modules>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.apache.camel</groupId>
        <artifactId>camel-bom</artifactId>
        <version>${version.camel}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
      <dependency>
        <groupId>org.apache.camel.kamelets</groupId>
        <artifactId>camel-kamelets</artifactId>
        <version>${version.camel-kamelets}</version>
      </dependency>
      <dependency>
        <groupId>org.apache.maven</groupId>
        <artifactId>maven-plugin-api</artifactId>
        <version>${version.maven-plugin-api}</version>
      </dependency>
      <dependency>
        <groupId>org.apache.maven.plugin-tools</groupId>
        <artifactId>maven-plugin-annotations</artifactId>
        <version>${version.maven-plugin-annotations}</version>
      </dependency>
      <dependency>
        <groupId>com.fasterxml.jackson.core</groupId>
        <artifactId>jackson-databind</artifactId>
        <version>${version.jackson}</version>
      </dependency>
      <dependency>
        <groupId>com.fasterxml.jackson.dataformat</groupId>
        <artifactId>jackson-dataformat-yaml</artifactId>
        <version>${version.jackson}</version>
      </dependency>
      <dependency>
        <groupId>io.fabric8</groupId>
        <artifactId>kubernetes-model</artifactId>
        <version>${version.kubernetes-model}</version>
      </dependency>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-api</artifactId>
        <version>${version.junit}</version>
        <scope>test</scope>
      </dependency>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-engine</artifactId>
        <version>${version.junit}</version>
        <scope>test</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <artifactId>maven-dependency-plugin</artifactId>
          <version>${version.maven-dependency-plugin}</version>
        </plugin>
        <plugin>
          <artifactId>maven-antrun-plugin</artifactId>
          <version>${version.maven-antrun-plugin}</version>
        </plugin>
        <plugin>
          <artifactId>maven-surefire-plugin</artifactId>
          <version>${version.maven-surefire-plugin}</version>
        </plugin>
        <plugin>
          <artifactId>maven-resources-plugin</artifactId>
          <version>${version.maven-resources-plugin}</version>
        </plugin>
        <plugin>
          <artifactId>maven-plugin-plugin</artifactId>
          <version>${version.maven-plugin-plugin}</version>
          <configuration>
            <encoding>${project.build.sourceEncoding}</encoding>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-compiler-plugin</artifactId>
          <version>${version.maven-compiler-plugin}</version>
          <configuration>
            <encoding>${project.build.sourceEncoding}</encoding>
            <release>${version.java}</release>
          </configuration>
        </plugin>
        <plugin>
          <groupId>org.apache.camel</groupId>
          <artifactId>camel-yaml-dsl-maven-plugin</artifactId>
          <version>${version.camel}</version>
        </plugin>
        <plugin>
          <groupId>io.kaoto</groupId>
          <artifactId>kaoto-camel-catalog-maven-plugin</artifactId>
          <version>${project.version}</version>
        </plugin>
      </plugins>
    </pluginManagement>
  </build>
</project>
