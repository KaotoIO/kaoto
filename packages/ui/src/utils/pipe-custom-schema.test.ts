import { cloneDeep } from 'lodash';
import { Pipe } from '@kaoto/camel-catalog/types';
import { updatePipeFromCustomSchema, getCustomSchemaFromPipe } from './';

describe('pipeCustomSchema', () => {
  let inputPipeStruct: Pipe;

  beforeEach(() => {
    inputPipeStruct = {
      apiVersion: 'camel.apache.org/v1',
      kind: 'Pipe',
      metadata: {
        name: 'webhook-binding',
        annotations: {
          'sco1237896.github.com/catalog.group': 'messaging',
          'sco1237896.github.com/catalog.id': 'http',
          'sco1237896.github.com/connector.description': 'Send data to a HTTP endpoint.',
          'sco1237896.github.com/connector.group': 'messaging',
          'sco1237896.github.com/connector.id': 'http_sink_v1',
          'sco1237896.github.com/connector.title': 'HTTP sink',
          'sco1237896.github.com/connector.version': 'v1',
          'trait.camel.apache.org/container.request-cpu': '0.20',
          'trait.camel.apache.org/container.request-memory': '128M',
          'trait.camel.apache.org/deployment.progress-deadline-seconds': '30',
          'trait.camel.apache.org/container.image': 'quay.io/sco1237896/connector-http:camel-4-1046a96',
        },
      },
      spec: {
        errorHandler: {
          log: {
            parameters: {
              maximumRedeliveries: 3,
              redeliveryDelay: 2000,
            },
          },
        },
        source: {
          ref: {
            kind: 'Kamelet',
            apiVersion: 'camel.apache.org/v1',
            name: 'webhook-source',
          },
        },
        steps: [
          {
            ref: {
              kind: 'Kamelet',
              apiVersion: 'camel.apache.org/v1',
              name: 'delay-action',
            },
          },
        ],
        sink: {
          ref: {
            kind: 'Kamelet',
            apiVersion: 'camel.apache.org/v1',
            name: 'log-sink',
          },
        },
      },
    };
  });

  describe('updatePipeFromCustomSchema', () => {
    it(`should preserve Pipe's original values when loading a malformed custom schema`, () => {
      const originalValue = cloneDeep(inputPipeStruct);
      const value = undefined;

      updatePipeFromCustomSchema(inputPipeStruct, value as unknown as Record<string, unknown>);
      expect(inputPipeStruct).toEqual(originalValue);
    });

    it('should mutate the original kamelet when loading a custom schema', () => {
      const value = {
        name: 'test',
        labels: {
          type: 'Action',
        },
        annotations: {
          foo: 'bar',
        },
      };

      updatePipeFromCustomSchema(inputPipeStruct, value as unknown as Record<string, unknown>);

      expect(inputPipeStruct.metadata!.name).toEqual(value.name);
      expect(inputPipeStruct.metadata!.labels).toEqual(value.labels);
      expect(inputPipeStruct.metadata!.annotations).toEqual(value.annotations);
    });
  });

  describe('getCustomSchemaFromPipe', () => {
    it('should get a custom pipe definition from a empty pipe', () => {
      const expectedCustomSchema = {
        name: '',
        labels: {},
        annotations: {},
      };

      const customSchema = getCustomSchemaFromPipe({} as Pipe);
      expect(customSchema).toEqual(expectedCustomSchema);
    });

    it('should get a custom pipe definition from a pipe official spec', () => {
      const expectedCustomSchema = {
        name: 'webhook-binding',
        labels: {},
        annotations: {
          'sco1237896.github.com/catalog.group': 'messaging',
          'sco1237896.github.com/catalog.id': 'http',
          'sco1237896.github.com/connector.description': 'Send data to a HTTP endpoint.',
          'sco1237896.github.com/connector.group': 'messaging',
          'sco1237896.github.com/connector.id': 'http_sink_v1',
          'sco1237896.github.com/connector.title': 'HTTP sink',
          'sco1237896.github.com/connector.version': 'v1',
          'trait.camel.apache.org/container.request-cpu': '0.20',
          'trait.camel.apache.org/container.request-memory': '128M',
          'trait.camel.apache.org/deployment.progress-deadline-seconds': '30',
          'trait.camel.apache.org/container.image': 'quay.io/sco1237896/connector-http:camel-4-1046a96',
        },
      };

      const customSchema = getCustomSchemaFromPipe(inputPipeStruct);
      expect(customSchema).toEqual(expectedCustomSchema);
    });
  });
});
