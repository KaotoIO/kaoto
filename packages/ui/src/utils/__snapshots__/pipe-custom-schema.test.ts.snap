// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeCustomSchema updatePipeFromCustomSchema should preserve Pipe's original values when loading a malformed custom schema 1`] = `
{
  "apiVersion": "camel.apache.org/v1",
  "kind": "Pipe",
  "metadata": {
    "annotations": {
      "sco1237896.github.com/catalog.group": "messaging",
      "sco1237896.github.com/catalog.id": "http",
      "sco1237896.github.com/connector.description": "Send data to a HTTP endpoint.",
      "sco1237896.github.com/connector.group": "messaging",
      "sco1237896.github.com/connector.id": "http_sink_v1",
      "sco1237896.github.com/connector.title": "HTTP sink",
      "sco1237896.github.com/connector.version": "v1",
      "trait.camel.apache.org/container.image": "quay.io/sco1237896/connector-http:camel-4-1046a96",
      "trait.camel.apache.org/container.request-cpu": "0.20",
      "trait.camel.apache.org/container.request-memory": "128M",
      "trait.camel.apache.org/deployment.progress-deadline-seconds": "30",
    },
    "labels": {},
    "name": "webhook-binding",
  },
  "spec": {
    "errorHandler": {
      "log": {
        "parameters": {
          "maximumRedeliveries": 3,
          "redeliveryDelay": 2000,
        },
      },
    },
    "sink": {
      "ref": {
        "apiVersion": "camel.apache.org/v1",
        "kind": "Kamelet",
        "name": "log-sink",
      },
    },
    "source": {
      "ref": {
        "apiVersion": "camel.apache.org/v1",
        "kind": "Kamelet",
        "name": "webhook-source",
      },
    },
    "steps": [
      {
        "ref": {
          "apiVersion": "camel.apache.org/v1",
          "kind": "Kamelet",
          "name": "delay-action",
        },
      },
    ],
  },
}
`;
